docs/concepts/README.md

```diff
@@ -0,0 +1,217 @@
+# Duck Typing
+
+Knative uses duck typing to keep various components loosely coupled, but
+[what is duck typing](https://en.wikipedia.org/wiki/Duck_typing)? It is
+reasoning about a resource's control plane shape and behaviors based on a common
+definition of that contact. If a resource has the same fields in the same schema
+locations as the contract specifies, and the control/data plane behaviors as the
+contract specifies, Knative can use that resource as if it is the generic duck
```

Comment:
```suggestion
common definition specifies, Knative can use that resource as if it is the generic duck
```

Reply:
I'd suggest changing this just because the term contract seems very abstract when we're already explaining some pretty abstract stuff, but if contract here means something specific feel free to ignore me